pcs/common/resource_agent_dto.py

```diff
@@ -61,12 +74,7 @@ class ResourceAgentParameterDto(DataTransferObject):
 
 @dataclass(frozen=True)
 class ResourceAgentMetadataDto(DataTransferObject):
-    # pylint: disable=too-many-instance-attributes
-
-    name: str  # full agent name: standard + provider + type
-    standard: str
-    provider: Optional[str]
-    type: str
+    agent: ResourceAgentNameDto
```

Comment:
Well, I wanted to avoid using `name` as it was used before. It won't make it backward compatible as originally I didn't plan to also change functions for conversion to the old format.